include/invert_preconditioner.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <invert_quda.h>
+#include <accelerator.h>
+
+#include <madwf_ml.h> // For MADWF
+
+namespace quda
+{
+  std::shared_ptr<Solver> create_preconditioner(const DiracMatrix &matPrecon, const DiracMatrix &matEig,
```

Comment:
Maybe outside the scope of this PR, but I think we should create a unified `interface` for when we have to create a preconditioner (what we do here) as opposed to when an existing one gets handed in (for multigrid --- see the two constructors for `GCR`). This would let us clean up some hackish boilerplate in the solver factory.

If anything we should make an issue for this---let me know if you think we should punt this to a separate PR (my MG-PCG work).

Reply:
See issue #1240.